tomchristie/django-rest-framework
# Version 3.5

* Requests Client
* API Client
* Hypermedia support
* API Client Uploads & Downloads
* RAML Support & Docs
* Schema fixes & Schema generation API
* Validation codes
* Technical error handlers
* Fix 1.10 deprecation warnings
* Add `HTML_SELECT_CUTOFF`, `HTML_SELECT_CUTOFF_TEXT` settings
* Add `SCHEMA_COERCE_PATH_PK`, `SCHEMA_COERCE_METHOD_NAMES` settings.

---

- [x] API Client - Credentials.
- [x] API Client - Headers.
- [x] API Client & Requests Client - Documentation.
- [x] API Client & Requests Client - ~~allow_external / reverse hosts etc~~.
- [ ] ~~Document client library uploads/downloads.~~ - Let's just refer to the more complete docs.
- [ ] Hypermedia support. (see #4241)
- [ ] Validation error codes. (see #3775)
- [ ] Finalize RAML support. (trailing slash, host name, docs, CSRF, login link, remove extra buttons, thank lynn root etc...)
- [ ] Review Swagger support - examples/types, basePath/not at root, examples. (see #4384)
- [ ] Review Django REST swagger integration.
- [ ] Document RAML support.
- [x] Schema fixes & new schema generation API.
- [x] Document schema generation API. (And limitations)
- [x] Formalize `get_schema_fields()` API.
- [ ] Document Cli, CoreJSON, CoreAPI.
- [ ] ~~Check remaining 1.10 deprecation warnings on tests.~~
- [x] Root node when using schemas (how to hyperlink?)
- [x] Schema view
- [x] Deprecation warnings (Both deal with those raised by Django 1.10, and move along any `PendingDeprecation`)
- [x] `pk` -> `id` in schema generation.
- [x] Media type for CoreJSON
- [x] Update schema examples. (New shortcut, and pk/id)
- [x] Deprecate Router init args for schemas.
- [ ] Technical error handlers
- [ ] ~~Register views with routers~~
- [x] Review #4401 "Schemas should automatically respect API root location"
